### ISSUE TITLE
Create new metrics Keras

### ISSUE THREAD

**Issue by tiagocalumbi** [fchollet/keras#4684]
Hi, guys!
How do I can create a new metric in Keras? 
I need to capture the values argmax of the y_true and y_predit, then,  I have to compare some values inside of the interval capture. 

I did without the variable Symbolic, but I have doubt when I use them.  Can anyone help? 

```
    interval = K.variable(50)
    k_true_max = K.argmax(th_true, axis=1)
    k_pred_max = K.argmax(th_pred, axis=1)

    label = th_true.eval()
    output = th_pred.eval()

    label = k_true_max.eval() #np.argmax(label, axis=1)
    output = k_pred_max.eval() #np.argmax(output, axis=1)

    sampleError = []

    print(label)
    print(output)
    for sampleIndex in range(len(label)):
        sampleLabel = label[sampleIndex]

        part_a = K.greater_equal(output[sampleIndex], sampleLabel - interval).eval() #a >= b
        part_b = K.lesser_equal(output[sampleIndex], sampleLabel + interval).eval()  #a <= b

        if(K.and_(part_a, part_b).eval()):
            sampleError.append(0)
        else:
            print(output[sampleIndex], sampleLabel)
            sampleError.append(1)

    print('result ', sampleError)
    print('mean ', np.mean(sampleError))

```

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by EncodeTS** [comment]
Is (this)[https://keras.io/metrics/#custom-metrics] what you want?

**Comment by tiagocalumbi** [comment]
Thank you for answering me.
I have the values of the y_true and y_pred with histograms example:
maybe I have a confidence interval "interval" 
interval = 2. 
y_true :  [[0,0,1,2,3,4,5,3,1],[0,0,0,0,0,1,2,5,2,1],[1,2,3,5,9,9,5,3,2,1],[0,0,3,5,7,8,7,5,3,0]]
y_pred:  [[0,0,2,4,4,7,4,1,2,0],[0,0,0,0,0,2,1,4,8,1],[0,6,0,1,2,1,4,0,0,0],[0,0,2,5,6,8,3,5,3,0]]

First step, I have to found the argmax of each array, also optimize the search.

label =   [5 6 4 5] #argmax de y_true
output = [4 8 1 6] #argmax de y_pred

Next to, I compare the values between in coordinates argmax with interval together. As result, I would have vector [1 1 0 1] within the mean them  been 75%
   
Now the problem! How I can assign the values symbolic theano? 
As I did, I 'm not sure whether the code's correct...because in the Keras the values is symbolics
